tools/cli/adminDBCommands.go

```diff
@@ -0,0 +1,264 @@
+	}
+	historyStore := cassp.NewHistoryV2PersistenceFromSession(session, loggerimpl.NewNopLogger())
+	branchDecoder := codec.NewThriftRWEncoder()
+	var token []byte
+	scanShardReport := &ScanShardReport{
+		ShardID: shardID,
+	}
+	isFirstIteration := true
+	for isFirstIteration || len(token) != 0 {
+		isFirstIteration = false
+		req := &persistence.ListConcreteExecutionsRequest{
+			PageSize:  executionPageSize,
+			PageToken: token,
+		}
+		resp, err := execStore.ListConcreteExecutions(req)
+		if err != nil {
+			writeToFile(scanFiles.failedToRunCheckFile, fmt.Sprintf("call to ListConcreteExecutions failed: %v", err))
+			scanShardReport.FailedToListExecutions = true
+			return scanShardReport
+		}
+		token = resp.NextPageToken
+
+		for _, e := range resp.ExecutionInfos {
+			scanShardReport.NumberOfExecutions++
+			var branch shared.HistoryBranch
+			err := branchDecoder.Decode(e.BranchToken, &branch)
+			if err != nil {
+				scanShardReport.NumberOfFailedChecks++
+				writeToFile(scanFiles.failedToRunCheckFile, fmt.Sprintf("failed to decode branch token: %v", err))
+				continue
+			}
+			readHistoryBranchReq := &persistence.InternalReadHistoryBranchRequest{
+				TreeID:    branch.GetTreeID(),
+				BranchID:  branch.GetBranchID(),
+				MinNodeID: 1,
+				MaxNodeID: 20,
+				ShardID:   shardID,
+				PageSize:  historyPageSize,
+			}
+			_, err = historyStore.ReadHistoryBranch(readHistoryBranchReq)
+			if err != nil {
+				if err == gocql.ErrNotFound {
+					metadata := ExecutionScanEntityMetadata{
+						Message:  "Detected workflow was corrupted based on missing history",
+						ErrorMsg: err.Error(),
+					}
+					scanShardReport.NumberOfCorruptedExecutions++
+					writeExecutionToFile(scanFiles.startEventCorruptFile, shardID, branch.GetTreeID(), branch.GetBranchID(), metadata, e)
+				} else {
+					metadata := ExecutionScanEntityMetadata{
+						Message:  "Checking corruption based on start event failed",
+						ErrorMsg: err.Error(),
+					}
+					scanShardReport.NumberOfFailedChecks++
+					writeExecutionToFile(scanFiles.failedToRunCheckFile, shardID, branch.GetTreeID(), branch.GetBranchID(), metadata, e)
+				}
+			}
+		}
+		time.Sleep(delayBetweenExecutionPages)
+	}
+	return scanShardReport
+}
+
+func writeToFile(file *os.File, message string) {
+	if _, err := file.WriteString(fmt.Sprintf("%v\r\n", message)); err != nil {
+		ErrorAndExit("failed to write to file", err)
+	}
+}
+
+func writeExecutionToFile(
+	file *os.File,
+	shardID int,
+	treeID string,
+	branchID string,
+	metadata ExecutionScanEntityMetadata,
+	info *persistence.InternalWorkflowExecutionInfo,
+) {
+	exec := ExecutionScanEntity{
+		ShardID:      shardID,
+		DomainID:     info.DomainID,
+		WorkflowID:   info.WorkflowID,
+		RunID:        info.RunID,
+		NextEventID:  info.NextEventID,
+		TreeID:       treeID,
+		BranchID:     branchID,
+		CloseStatus:  info.CloseStatus,
+		ScanMetadata: metadata,
+	}
+	data, err := json.Marshal(exec)
```

Comment:
Probably use `MarshalIndent` to make it more readable. Same for other write function.